File: c++/22945.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

vector<int> v;

int main() {
  freopen("example.txt", "r", stdin);

  int N;
  cin >> N;

  while (N--) {
    int a;
    cin >> a;
    v.push_back(a);
  }

  int i = 0;
  int j = v.size() - 1;
  int ans = 0;

  while (i < j) {
    int k = min(v[i], v[j]) * (j - i - 1);
    ans = max(k, ans);

    if (v[i] < v[j])
      i++;
    else
      j--;
  }

  cout << ans;

  return 0;
}

File: js/22945.js
const fs = require('fs');

const inputs = fs.readFileSync('example.txt').toString().trim().split('\n');

const arr = inputs[1].split(' ').map(Number);

let i = 0;
let j = arr.length - 1;

let ans = 0;

while (i < j) {
  const k = Math.min(arr[i], arr[j]) * (j - i - 1);

  ans = Math.max(ans, k);

  if (arr[i] < arr[j]) i++;
  else j--;
}

console.log(ans);
